anaDACScan.py

```diff
@@ -0,0 +1,321 @@
+    
+    dataPath = os.getenv("DATA_PATH")
+    elogPath = os.getenv("ELOG_PATH") 
+    
+    if len(args.infilename.split('/')) > 1 and len(args.infilename.split('/')[len(args.infilename.split('/')) - 2].split('.')) == 5:
+        scandate = args.infilename.split('/')[len(args.infilename.split('/')) - 2]
+    else:    
+        scandate = 'noscandate'
+        
+    for oh in ohArray:
+        if scandate == 'noscandate':
+            os.system("mkdir -p "+ elogPath+"/"+chamber_config[oh])
+        else:
+            os.system("mkdir -p "+ dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate)
+            
+    # Determine which DAC was scanned and against which ADC
+    nameX = ""
+    nameY = ""
+    for event in dacScanFile.dacScanTree:
+        nameX = str(event.nameX.data())
+        nameY = str(event.nameY.data())
+        break # all entries will be the same
+
+    if nameY not in ['ADC0', 'ADC1']:
+        print("Error: unexpected value of nameY: '%s'"%nameY)
+        exit(1)
+
+    from utils.anaInfo import nominalDacValues
+    from utils.anaInfo import nominalDacScalingFactors
+        
+    if nameX not in nominalDacValues.keys():
+        print("Error: unexpected value of nameX: '%s'"%nameX)
+        exit(1)
+
+    #the nominal reference current is 10 uA and it has a scaling factor of 0.5   
+    nominal_iref = 10/0.5
+
+    nominal = nominalDacValues[nameX][0]/nominalDacScalingFactors[nameX]
+
+    #convert all voltages to mV and currents to uA
+    if nominalDacValues[nameX][1] == "V":
+        nominal *= pow(10.0,3)
+    elif nominalDacValues[nameX][1] == 'mV':
+        pass
+    elif nominalDacValues[nameX][1] == 'uV':
+        nominal *= pow(10.0,-3)
+    elif nominalDacValues[nameX][1] == 'nV':
+        nominal *= pow(10.0,-6)
+    elif nominalDacValues[nameX][1] == "A":
+        nominal *= pow(10.0,6)
+    elif nominalDacValues[nameX][1] == 'mA':
+        nominal *= pow(10.0,3)
+    elif nominalDacValues[nameX][1] == 'uA':
+        pass
+    elif nominalDacValues[nameX][1] == 'nA':
+        nominal *= pow(10.0,-3)
+    else:
+        print("Error: unexpected units: '%s'"%nominalDacValues[nameX][1])
+        exit(1)
+
+    calInfo = {}
+
+    if args.calFileList != None:
+        for line in open(args.calFileList):
+            if line[0] == "#":
+                continue
+            line = line.strip(' ').strip('\n')
+            first = line.split(' ')[0].strip(' ')
+            second = line.split(' ')[1].strip(' ')
+            tuple_calInfo = parseCalFile(second)
+            calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+
+    #for each OH, check if calibration files were provided, if not search for the calFile in the $DATA_PATH, if it is not there, then skip that OH for the rest of the script
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calAdcCalFile = "{0}/{1}/calFile_calAdc_{1}.txt".format(dataPath,chamber_config[oh])
```

Comment:
This doesn't match the previous comment, this should explicitly be what was shown [here](https://github.com/cms-gem-daq-project/gem-plotting-tools/pull/158#discussion_r225479887).

Reply:
done